src/lib.rs

```diff
@@ -97,31 +97,19 @@ impl<R: Runtime> WindowExt for Window<R> {
     if let Some(state) = c.get(self.label()) {
       self.set_decorations(state.decorated)?;
 
-      let mut pos: Option<(i32, i32)> = None;
-      for m in self.available_monitors()? {
-        if m.name().map(ToString::to_string).unwrap_or_default() == state.monitor {
-          pos = Some((state.x, state.y));
-          break;
-        }
-      }
-
       self.set_size(LogicalSize {
         width: state.width,
         height: state.height,
       })?;
 
-      let (x, y) = match pos {
-        Some((x, y)) => (x, y),
-        None => {
-          if let Some(m) = self.current_monitor()? {
-            let mpos = m.position().to_logical::<i32>(m.scale_factor());
-            (mpos.x + 100, mpos.y + 100)
-          } else {
-            (100, 100)
-          }
+      // restore position to saved value if saved monitor exists
+      //  otherwise, let the OS decide where to place the window
+      for m in self.available_monitors()? {
+        if m.name().map(ToString::to_string).unwrap_or_default() == state.monitor {
+          self.set_position(PhysicalPosition { x: state.x, y: state.y })?;
+          break;
         }
-      };
-      self.set_position(LogicalPosition { x, y })?;
+      }
```

Comment:
Really good, if we don't find the last saved monitor, it defaults to the OS position.